DatadogSessionReplay/Sources/Recorder/Recorder.swift

```diff
@@ -119,37 +109,19 @@ public class Recorder: Recording {
 
     /// Initiates the capture of a next record.
     /// **Note**: This is called on the main thread.
-    func captureNextRecord(_ recorderContext: Context) {
-        let methodCalledTrace = telemetry.startMethodCalled(
-            operationName: MethodCallConstants.captureRecordOperationName,
-            callerClass: MethodCallConstants.className,
-            samplingRate: methodCallTelemetrySamplingRate // Effectively 3% * 0.1% = 0.003% of calls
-        )
-        var isSuccessful = true
-        do {
-            guard let viewTreeSnapshot = try viewTreeSnapshotProducer.takeSnapshot(with: recorderContext) else {
-                // There is nothing visible yet (i.e. the key window is not yet ready).
-                return
-            }
-            let touchSnapshot = touchSnapshotProducer.takeSnapshot(context: recorderContext)
-            snapshotProcessor.process(viewTreeSnapshot: viewTreeSnapshot, touchSnapshot: touchSnapshot)
-
-            resourceProcessor.process(
-                resources: viewTreeSnapshot.resources,
-                context: .init(recorderContext.applicationID)
-            )
-        } catch let error {
-            isSuccessful = false
-            telemetry.error("[SR] Failed to take snapshot", error: DDError(error: error))
+    func captureNextRecord(_ recorderContext: Context) throws {
+        guard let viewTreeSnapshot = try viewTreeSnapshotProducer.takeSnapshot(with: recorderContext) else {
```

Comment:
question/ Won't the Swift error thrown from here be wrapped into `ObjcException` and falsely reported as Obj-c issue to telemetry? Maybe we should split this call into two succeeding calls and pass the `viewTreeSnapshot` result to `captureNextRecord(_:viewTreeSnapshot:)`?

Reply:
No, because `objc_rethrow` implementation catches Swit errors and [rethrow them as-is](https://github.com/DataDog/dd-sdk-ios/blob/f824015317e2f3f7a01fa3578b94983ed39c1512/DatadogInternal/Sources/Utils/DDError.swift#L136).
I will add test coverage on this 👍